php-frontend/src/main/java/org/sonar/php/PHPAnalyzer.java

```diff
@@ -102,20 +103,17 @@ public void nextFile(PhpFile file) {
         LOG.error("Stack overflow of {} in file {}", check.getClass().getName(), currentFile.uri());
         throw e;
       }
-      issuesBuilder.addAll(issues);
+      allIssues.addAll(issues);
     }
-
-    return issuesBuilder.build();
+    return allIssues;
   }
 
   public List<PhpIssue> analyzeTest() {
-    ImmutableList.Builder<PhpIssue> issuesBuilder = ImmutableList.builder();
-    for (PHPCheck check : testFileChecks) {
-      PHPCheckContext context = new PHPCheckContext(currentFile, currentFileTree, workingDir, currentFileSymbolTable);
-      issuesBuilder.addAll(check.analyze(context));
-    }
-
-    return issuesBuilder.build();
+    PHPCheckContext context = new PHPCheckContext(currentFile, currentFileTree, workingDir, currentFileSymbolTable);
+    return testFileChecks.stream()
+      .map(check -> check.analyze(context))
+      .flatMap(List::stream)
+      .collect(Collectors.toList());
```

Comment:
Now, a single `context` is shared by all checks. It's probably not a big deal but I don't really like having that change in the current PR.